smarthome.cpp

```diff
@@ -0,0 +1,82 @@
+#include "smarthome.h"
+#include <fstream>
+#include <iostream>
+
+SmartHome::SmartHome(const std::vector<Room> rooms)
+    : rooms(rooms) {}
+
+const std::vector<Room> SmartHome::getRooms() const {
+    return rooms;
+}
+
+void SmartHome::readDataFromFile(const std::string& filename) {
+    std::ifstream file(filename);
+
+    if (!file.is_open()) {
+        std::cerr << "Error: Unable to open file " << filename << std::endl;
+        return;
+    }
+
+    rooms.clear(); // Curățăm vectorul existent înainte de a încărca noi date
```

Comment:
not really important, but be consistent with the comments language(English preferred)